Rust client doesn't compile

`cargo check` output:
```
error[E0106]: missing lifetime specifier
  --> clients/rust/src/generated/instructions/add_memo.rs:53:15
   |
53 |     pub memo: &str,
   |               ^ expected named lifetime parameter
   |
help: consider introducing a named lifetime parameter
   |
52 ~ pub struct AddMemoInstructionArgs<'a> {
53 ~     pub memo: &'a str,
   |

error[E0637]: `&` without an explicit lifetime name cannot be used here
  --> clients/rust/src/generated/instructions/add_memo.rs:53:15
   |
53 |     pub memo: &str,
   |               ^ explicit lifetime name needed here
   |
help: consider introducing a higher-ranked lifetime here
   |
50 ~ #[derive(for<'a> BorshSerialize, BorshDeserialize, Clone, Debug, Eq, PartialEq)]
51 | #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
52 | pub struct AddMemoInstructionArgs {
53 ~     pub memo: &'a str,
   |

error[E0637]: `&` without an explicit lifetime name cannot be used here
  --> clients/rust/src/generated/instructions/add_memo.rs:53:15
   |
53 |     pub memo: &str,
   |               ^ explicit lifetime name needed here
   |
help: consider introducing a higher-ranked lifetime here
   |
50 ~ #[derive(BorshSerialize, for<'a> BorshDeserialize, Clone, Debug, Eq, PartialEq)]
51 | #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
52 | pub struct AddMemoInstructionArgs {
53 ~     pub memo: &'a str,
   |

error[E0106]: missing lifetime specifier
  --> clients/rust/src/generated/instructions/add_memo.rs:62:18
   |
62 |     memo: Option<&str>,
   |                  ^ expected named lifetime parameter
   |
help: consider introducing a named lifetime parameter
   |
61 ~ pub struct AddMemoBuilder<'a> {
62 ~     memo: Option<&'a str>,
   |

error[E0106]: missing lifetime specifier
   --> clients/rust/src/generated/instructions/add_memo.rs:265:18
    |
265 |     memo: Option<&str>,
    |                  ^ expected named lifetime parameter
    |
note: these named lifetimes are available to use
   --> clients/rust/src/generated/instructions/add_memo.rs:263:37
    |
263 | struct AddMemoCpiBuilderInstruction<'a, 'b> {
    |                                     ^^  ^^
help: consider using one of the available lifetimes here
    |
265 |     memo: Option<&'lifetime str>,
    |                   +++++++++

error[E0277]: the trait bound `&str: BorshDeserialize` is not satisfied
  --> clients/rust/src/generated/instructions/add_memo.rs:50:26
   |
50 | #[derive(BorshSerialize, BorshDeserialize, Clone, Debug, Eq, PartialEq)]
   |                          ^^^^^^^^^^^^^^^^ the trait `BorshDeserialize` is not implemented for `&str`
   |
   = help: the trait `BorshDeserialize` is implemented for `String`
   = help: see issue #48214
   = note: this error originates in the derive macro `BorshDeserialize` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0106, E0277, E0637.
For more information about an error, try `rustc --explain E0106`.
error: could not compile `solana-memo` (lib) due to 6 previous errors
(base) user@mr-tuxedoaura15gen3:~/memo/clients/rust$ cargo check
    Checking solana-memo v0.0.0 (/home/user/memo/clients/rust)
error[E0106]: missing lifetime specifier
  --> clients/rust/src/generated/instructions/add_memo.rs:53:15
   |
53 |     pub memo: &str,
   |               ^ expected named lifetime parameter
   |
help: consider introducing a named lifetime parameter
   |
52 ~ pub struct AddMemoInstructionArgs<'a> {
53 ~     pub memo: &'a str,
   |

error[E0637]: `&` without an explicit lifetime name cannot be used here
  --> clients/rust/src/generated/instructions/add_memo.rs:53:15
   |
53 |     pub memo: &str,
   |               ^ explicit lifetime name needed here
   |
help: consider introducing a higher-ranked lifetime here
   |
50 ~ #[derive(for<'a> BorshSerialize, BorshDeserialize, Clone, Debug, Eq, PartialEq)]
51 | #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
52 | pub struct AddMemoInstructionArgs {
53 ~     pub memo: &'a str,
   |

error[E0637]: `&` without an explicit lifetime name cannot be used here
  --> clients/rust/src/generated/instructions/add_memo.rs:53:15
   |
53 |     pub memo: &str,
   |               ^ explicit lifetime name needed here
   |
help: consider introducing a higher-ranked lifetime here
   |
50 ~ #[derive(BorshSerialize, for<'a> BorshDeserialize, Clone, Debug, Eq, PartialEq)]
51 | #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
52 | pub struct AddMemoInstructionArgs {
53 ~     pub memo: &'a str,
   |

error[E0106]: missing lifetime specifier
  --> clients/rust/src/generated/instructions/add_memo.rs:62:18
   |
62 |     memo: Option<&str>,
   |                  ^ expected named lifetime parameter
   |
help: consider introducing a named lifetime parameter
   |
61 ~ pub struct AddMemoBuilder<'a> {
62 ~     memo: Option<&'a str>,
   |

error[E0106]: missing lifetime specifier
   --> clients/rust/src/generated/instructions/add_memo.rs:265:18
    |
265 |     memo: Option<&str>,
    |                  ^ expected named lifetime parameter
    |
note: these named lifetimes are available to use
   --> clients/rust/src/generated/instructions/add_memo.rs:263:37
    |
263 | struct AddMemoCpiBuilderInstruction<'a, 'b> {
    |                                     ^^  ^^
help: consider using one of the available lifetimes here
    |
265 |     memo: Option<&'lifetime str>,
    |                   +++++++++

error[E0277]: the trait bound `&str: BorshDeserialize` is not satisfied
  --> clients/rust/src/generated/instructions/add_memo.rs:50:26
   |
50 | #[derive(BorshSerialize, BorshDeserialize, Clone, Debug, Eq, PartialEq)]
   |                          ^^^^^^^^^^^^^^^^ the trait `BorshDeserialize` is not implemented for `&str`
   |
   = help: the trait `BorshDeserialize` is implemented for `String`
   = help: see issue #48214
   = note: this error originates in the derive macro `BorshDeserialize` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0106, E0277, E0637.
For more information about an error, try `rustc --explain E0106`.
error: could not compile `solana-memo` (lib) due to 6 previous errors
```

--- a/clients/rust/Cargo.toml
+++ b/clients/rust/Cargo.toml
@@ -11,6 +11,7 @@ serde = ["dep:serde", "dep:serde_with"]
 
 [dependencies]
 borsh = "^0.10"
+kaigan = "0.2.5"
 num-derive = "^0.3"
 num-traits = "^0.2"
 serde = { version = "^1.0", features = ["derive"], optional = true }

--- a/clients/rust/src/generated/instructions/add_memo.rs
+++ b/clients/rust/src/generated/instructions/add_memo.rs
@@ -7,6 +7,7 @@
 
 use borsh::BorshDeserialize;
 use borsh::BorshSerialize;
+use kaigan::types::RemainderStr;
 
 /// Accounts.
 pub struct AddMemo {}
@@ -39,10 +40,10 @@ impl AddMemo {
 }
 
 #[derive(BorshDeserialize, BorshSerialize)]
-struct AddMemoInstructionData {}
+pub struct AddMemoInstructionData {}
 
 impl AddMemoInstructionData {
-    fn new() -> Self {
+    pub fn new() -> Self {
         Self {}
     }
 }
@@ -50,7 +51,7 @@ impl AddMemoInstructionData {
 #[derive(BorshSerialize, BorshDeserialize, Clone, Debug, Eq, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct AddMemoInstructionArgs {
-    pub memo: &str,
+    pub memo: RemainderStr,
 }
 
 /// Instruction builder for `AddMemo`.
@@ -59,7 +60,7 @@ pub struct AddMemoInstructionArgs {
 ///
 #[derive(Default)]
 pub struct AddMemoBuilder {
-    memo: Option<&str>,
+    memo: Option<RemainderStr>,
     __remaining_accounts: Vec<solana_program::instruction::AccountMeta>,
 }
 
@@ -68,7 +69,7 @@ impl AddMemoBuilder {
         Self::default()
     }
     #[inline(always)]
-    pub fn memo(&mut self, memo: &str) -> &mut Self {
+    pub fn memo(&mut self, memo: RemainderStr) -> &mut Self {
         self.memo = Some(memo);
         self
     }
@@ -201,7 +202,7 @@ impl<'a, 'b> AddMemoCpiBuilder<'a, 'b> {
         Self { instruction }
     }
     #[inline(always)]
-    pub fn memo(&mut self, memo: &str) -> &mut Self {
+    pub fn memo(&mut self, memo: RemainderStr) -> &mut Self {
         self.instruction.memo = Some(memo);
         self
     }
@@ -262,7 +263,7 @@ impl<'a, 'b> AddMemoCpiBuilder<'a, 'b> {
 
 struct AddMemoCpiBuilderInstruction<'a, 'b> {
     __program: &'b solana_program::account_info::AccountInfo<'a>,
-    memo: Option<&str>,
+    memo: Option<RemainderStr>,
     /// Additional instruction accounts `(AccountInfo, is_writable, is_signer)`.
     __remaining_accounts: Vec<(
         &'b solana_program::account_info::AccountInfo<'a>,

--- a/package.json
+++ b/package.json
@@ -11,7 +11,7 @@
   },
   "devDependencies": {
     "@iarna/toml": "^2.2.5",
-    "@metaplex-foundation/kinobi": "^0.18.1",
+    "@metaplex-foundation/kinobi": "^0.18.5",
     "@metaplex-foundation/shank-js": "^0.1.7",
     "typescript": "^5.4.2",
     "zx": "^7.2.3"

